src/ranged.cpp

```diff
@@ -1845,12 +1851,14 @@ static void cycle_action( item &weap, const tripoint &pos )
         cargo = vp->vehicle().get_parts_at( pos, "CARGO", part_status_flag::any );
     }
 
+    item &parent = weap.parent_item() != nullptr &&
+                   weap.has_flag( flag_USE_PARENT_GUN ) ? *weap.parent_item() : weap;
```

Comment:
```suggestion
    item &parent = ( weap.parent_item() != nullptr &&
                   weap.has_flag( flag_USE_PARENT_GUN ) ) ? *weap.parent_item() : weap;
```
and again. Since this is the third use of the same phrase it might be useful to make it its own function. Something like
```cpp
static item &get_item_parent_or_fail( item &it ) {
    return ( it.parent_item() != nullptr && it.has_flag( flag_USE_PARENT_GUN ) ? *it.parent_item() : it;
}
```
and replace my suggestions with `item &parent = get_item_parent_or_fail( whatever_the_item_is_called_here );`